docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+- hello.wasm, the WebAssembly binary file for the smart contract.
+- hello.abi, the application binary interface (ABI) file for the smart contract.
+
+#### Deploy the hello Smart Contract (built with cmake)
+
+Execute the following command to deploy the `hello` smart contract to the account `hello` which you created at the beginning:
+
+```shell
+dune --deploy <PATH_TO_CMakeLists.txt_PARENT_DIR>/build/hello
+```
+
+## First Test
+
+Let's send the `hi` action to the local node and send as input parameter the test account name you created earlier:
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+The output of the command above shows on the first line that `hello::hi` action was executed with the input parameter `{"nm":"ama"}` and on the second line the output of the action itself `Name: ama`.
+
+```txt
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name: ama
+```
+
+## Smart Contract Source Files
+
+The smart contract C++ source files are:
+
+- hello.hpp
+- hello.cpp
+
+### The hello.hpp File
+
+The `hello.hpp` is the C++ header file and it contains the `hello` smart contract C++ class definition.
+
+```c++
+#include <eosio/eosio.hpp>
+using namespace eosio;
+
+CONTRACT hello : public contract {
+   public:
+      using contract::contract;
+
+      ACTION hi( name nm );
+};
+```
+
+A smart contract class definition must:
+
+- Be annotated by the [[eosio::contract]] attribute which tells the compiler it is a smart contract class; in the `hello.hpp` generated code the `CONTRACT` macro is used, which expands to `class [[eosio::contract]]` C++ code.
+- Be derived from the `contract` class which provides basic smart contract functionality.
+- Define at least a public action function, in this case it has only one, the `hi` action.
+- Define a public action wrapper for each action defined.
+
+You will learn more about actions later.
+
+### The hello.cpp File
+
+The `hello.cpp` is the C++ implementation file and it contains the `hello` smart contract C++ class implementation, that is, the implementation for every member function of the class.
+
+## Actions
+
+An action is a method, defined and implemented by a smart contract class, that can be called by an authorized account to perform some specific task or operation on the blockchain. Actions can be called by other contracts or by external accounts with the EOS Chain API.
+
+Actions can have parameters and return values and their responsibility is to execute the business logic of the contract. For example, an action in a token contract might transfer tokens from one account to another or update the balances of multiple accounts.
+
+Each action requires a specific level of authorization, which can be specified in the action's code. This ensures that only authorized accounts can execute specific actions.
+
+The `hello` smart contract class has only one action implemented by its `hi` public member function.
```

Comment:
Said before above

Reply:
I am not sure what you mean; above we show the declaration of the action, here we show its implementation.

Reply:
ok got it.